demos/2d_cylindrical/2d_cylindrical.py

```diff
@@ -110,8 +113,8 @@
     f_ratio = rmin/rmax
     top_scaling = -1.3290170684486309  # log(f_ratio) / (1.- f_ratio)
     bot_scaling = -0.7303607313096079  # (f_ratio * log(f_ratio)) / (1.- f_ratio)
-    nusselt_number_top = (assemble(dot(grad(T), n) * ds_t) / assemble(Constant(1.0, domain=mesh)*ds_t)) * top_scaling
-    nusselt_number_base = (assemble(dot(grad(T), n) * ds_b) / assemble(Constant(1.0, domain=mesh)*ds_b)) * bot_scaling
+    nusselt_number_top = (assemble(dot(grad(T), n) * ds_t) / top_area) * top_scaling
+    nusselt_number_base = (assemble(dot(grad(T), n) * ds_b) / bottom_area) * bot_scaling
```

Comment:
Or maybe we can just remove `domain=mesh` from the Constant without penalty? It does seem a bit dumb to have to use Functions to do it...